src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs

```diff
@@ -4,22 +4,25 @@
 using System;
 using System.Diagnostics;
 using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Settings;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
 {
     public class SystemTraceWriter : TraceWriter
     {
         private IEventGenerator _eventGenerator;
+        private ISettingsManager _settingsManager;
         private string _appName;
         private string _subscriptionId;
-
-        public SystemTraceWriter(TraceLevel level) : this(new EventGenerator(), level)
+        
+        public SystemTraceWriter(ISettingsManager settingsManager, TraceLevel level) : this(new EventGenerator(), settingsManager, level)
```

Comment:
The intent for this ctor is to default the params. Can't you default the SettingsManager as well (like EventGenerator)?